MediaBrowser.Providers/Playlists/PlaylistItemsProvider.cs

```diff
@@ -59,109 +63,127 @@ public Task<ItemUpdateType> FetchAsync(Playlist item, MetadataRefreshOptions opt
         {
-            var content = new M3uContent();
+            var content = new ZplContent();
             var playlist = content.GetFromStream(stream);
 
-            return playlist.PlaylistEntries.Select(i => new LinkedChild
-            {
-                Path = GetPlaylistItemPath(i.Path, path),
-                Type = LinkedChildType.Manual
-            });
+            return playlist.PlaylistEntries
+                    .Select(i => GetLinkedChild(i.Path, playlistPath, libraryRoots))
+                    .Where(i => i is not null);
         }
 
-        private IEnumerable<LinkedChild> GetZplItems(Stream stream, string path)
+        private IEnumerable<LinkedChild> GetWplItems(Stream stream, string playlistPath, List<string> libraryRoots)
         {
-            var content = new ZplContent();
+            var content = new WplContent();
             var playlist = content.GetFromStream(stream);
 
-            return playlist.PlaylistEntries.Select(i => new LinkedChild
-            {
-                Path = GetPlaylistItemPath(i.Path, path),
-                Type = LinkedChildType.Manual
-            });
+            return playlist.PlaylistEntries
+                    .Select(i => GetLinkedChild(i.Path, playlistPath, libraryRoots))
+                    .Where(i => i is not null);
         }
 
-        private IEnumerable<LinkedChild> GetWplItems(Stream stream, string path)
+        private LinkedChild GetLinkedChild(string itemPath, string playlistPath, List<string> libraryRoots)
         {
-            var content = new WplContent();
-            var playlist = content.GetFromStream(stream);
-
-            return playlist.PlaylistEntries.Select(i => new LinkedChild
+            if (TryGetPlaylistItemPath(itemPath, playlistPath, libraryRoots, out var parsedPath))
             {
-                Path = GetPlaylistItemPath(i.Path, path),
-                Type = LinkedChildType.Manual
-            });
+                return new LinkedChild
+                {
+                    Path = parsedPath,
+                    Type = LinkedChildType.Manual
+                };
+            }
+
+            return null;
         }
 
-        private string GetPlaylistItemPath(string itemPath, string containingPlaylistFolder)
+        private bool TryGetPlaylistItemPath(string itemPath, string playlistPath, List<string> libraryPaths, out string path)
         {
-            if (!File.Exists(itemPath))
+            path = null;
+            string pathToCheck;
+            if (File.Exists(itemPath))
+            {
+                pathToCheck = itemPath;
+            }
+            else
+            {
+                var baseFolder = Path.GetDirectoryName(playlistPath);
+                var basePath = Path.Combine(baseFolder, itemPath);
+                pathToCheck = Path.GetFullPath(basePath);
+                if (!File.Exists(pathToCheck))
+                {
+                    return false;
+                }
+            }
```

Comment:
The whole logic may be simplified by `BaseItem.FileSystem.MakeAbsolutePath()`. That method has some extensive check of relative path, and returns the absolute path. You pass in a containing folder and a path to be checked, if the path to be checked is absolute, it will return it directly, otherwise it will combine both paths to make it absolute.